itest/rust/src/framework/runner.rs

```diff
@@ -191,36 +318,36 @@ impl IntegrationTests {
                 }
             };
 
-            self.update_stats(&outcome, &test_file, &test_case);
+            Self::update_stats(&mut self.stats, &outcome, &test_file, &test_case);
```

Comment:
Why this change, is this due to some partial borrowing shenanigans?

Reply:
Yes `update_stats` can't access `&mut self` anymore in the async setup, but in both cases we can just pass in the stats' struct.